File: multiscale_benchmark/2022_09_hackathon/Biodynamo/unit_test_cellcycle/new_results/postprocess.m
clear
clc
%close all

data = table2array(readtable('cell-0-sim1.csv'));

Time = data(:,1);
Volume = data(:,2);
Phase = data(:,3);
Age = data(:,4);

Volume_min = Volume(1);

figure, hold on
plot(Time, Volume/Volume_min, '-r')
xlabel('time (hours)'), ylabel('cell volume (normalized to minimum value)')
grid on
hold off

figure, hold on
plot(Time, Phase, '-b')
xlabel('time (hours)'), ylabel('cell phase (1:G0/G1; 2:S; 3:G2; 4:M)')
grid on
hold off

File: multiscale_benchmark/2022_09_hackathon/Biodynamo/unit_test_cellcycle_stoch/new_results/postprocess.py
import pandas as pd
import matplotlib.pyplot as plt

data = pd.read_csv('cell-0-sim1.csv')

data.columns = ['Time', 'Volume', 'Phase', 'Age']

Time = data['Time']
Volume = data['Volume']
Phase = data['Phase']
Age = data['Age']

Volume_min = Volume[0]

plt.figure()
plt.plot(Time, Volume/Volume_min, '-r')
plt.xlabel('time (hours)')
plt.ylabel('cell volume (normalized to minimum value)')
plt.grid(True)
plt.savefig('VolumeVsTime.png')

plt.figure()
plt.plot(Time, Phase, '-b')
plt.xlabel('time (hours)')
plt.ylabel('cell phase (1:G0/G1; 2:S; 3:G2; 4:M)')
plt.grid(True)
plt.savefig('PhaseVsTime.png')